Testscripts/Windows/VERIFY-XDP-MULTIPLE-NICS.ps1

```diff
@@ -0,0 +1,195 @@
+            $timer = 0
+            while ((Get-Job -Id $testJob_1).State -eq "Running" -or (Get-Job -Id $testJob_2).State -eq "Running") {
+                $currentStatus = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort -username $user -password $password `
+                    -command "tail -2 ~/xdpdumpout_eth1.txt | head -1" -runAsSudo
+                Write-LogInfo "Current Test Status: eth1 : $currentStatus"
+                $currentStatus = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort -username $user -password $password `
+                    -command "tail -2 ~/xdpdumpout_eth2.txt | head -1" -runAsSudo
+                Write-LogInfo "Current Test Status: eth2 : $currentStatus"
+                Wait-Time -seconds 20
+                $timer += 1
+                if ($timer -gt 15) {
+                    Throw "XDPDump did not stop after 5 mins. Please check logs"
+                }
+            }
+
+            $currentStatus_1 = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort -username $user -password $password `
+                -command "tail -1 ~/xdpdumpout_eth1.txt" -runAsSudo
+            $currentStatus_2 = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort -username $user -password $password `
+                -command "tail -1 ~/xdpdumpout_eth2.txt" -runAsSudo
+            if ( ($currentStatus_1 -match "unloading xdp") -and ($currentStatus_2 -match "unloading xdp") ) {
+                $testResult = "PASS"
+            }
+            else {
+                Write-LogErr "Test Aborted. Last known status : $currentStatus_1 & $currentStatus_2."
+                Throw "XDPDump aborted. please check logs"
+            }
+            Write-LogInfo "Successfully ran xdpdump on both the nics"
+
+        }   elseif ($currentState -imatch "TestAborted") {
+            Write-LogErr "Test Aborted. Last known status: $currentStatus."
+            $testResult = "ABORTED"
+        }   elseif ($currentState -imatch "TestSkipped") {
+            Write-LogErr "Test Skipped. Last known status: $currentStatus"
+            $testResult = "SKIPPED"
+        }	elseif ($currentState -imatch "TestFailed") {
+            Write-LogErr "Test failed. Last known status: $currentStatus."
+            $testResult = "FAIL"
+        }	else {
+            Write-LogErr "Test execution is not successful, check test logs in VM."
+            $testResult = "ABORTED"
+        }
+
+        Copy-RemoteFiles -downloadFrom $clientVMData.PublicIP -port $clientVMData.SSHPort `
+            -username $user -password $password -download `
+            -downloadTo $LogDir -files "*.txt, *.log"
+    }
+    catch {
+        $ErrorMessage = $_.Exception.Message
+        $ErrorLine = $_.InvocationInfo.ScriptLineNumber
+        Write-LogErr "EXCEPTION : $ErrorMessage at line: $ErrorLine"
+        $testResult = "FAIL"
+    }
+    finally {
+        if (!$testResult) {
+            $testResult = "ABORTED"

```

Comment:
What condition does it reach to this line?

Reply:
This condition makes sure that we don't return null value.

Reply:
No, I meant this assignment statement never exercise at all. Any exception goes to catch block and get the value "FAIL", and finally block always has testResult value. Am I missing here?

Reply:
I like to keep a check on values that are returned.